### ISSUE TITLE
Install: gnomad for hg19 fails

### ISSUE THREAD

**Issue by lbeltrame** [bcbio/bcbio-nextgen#2816]
After downloading, this error occurs after some time (~30-40 min):

```
> gsort version 0.0.6
uniq v0.57

options:     input VCF file        -normalize v
0.5         [o] output VCF file       
-

options:     input VCF file                                  -
         [o] output VCF file                                 -
         [w] sorting window size                             10000
         [n] no fail on reference inconsistency for non SNPs true
         [q] quiet                                           false
         [d] debug                                           false
         [r] reference FASTA file                            ../seq/hg19.fa

[W::vcf_parse] Contig 'chr1' is not defined in the header. (Quick workaround: index the file with tabix.)
Warning: The tag "AC_nfe_bgr" not defined in the header
Warning: The tag "AN_nfe_bgr" not defined in the header
Warning: The tag "AF_nfe_bgr" not defined in the header
Warning: The tag "nhomalt_nfe_bgr" not defined in the header
Warning: The tag "AC_sas_male" not defined in the header
Warning: The tag "AN_sas_male" not defined in the header
Warning: The tag "AF_sas_male" not defined in the header
[W::vcf_parse] Contig 'chr1' is not defined in the header. (Quick workaround: index the file with tabix.)
Encountered error, cannot proceed. Please check the error output above.

stats: biallelic
          no. left trimmed                      : 0
          no. right trimmed                     : 0
          no. left and right trimmed            : 0
          no. right trimmed and left aligned    : 0
          no. left aligned                      : 0

       total no. biallelic normalized           : 0

       multiallelic
          no. left trimmed                      : 0
          no. right trimmed                     : 0
          no. left and right trimmed            : 0
          no. right trimmed and left aligned    : 0
          no. left aligned                      : 0

       total no. multiallelic normalized        : 0

       total no. variants normalized            : 0
       total no. variants observed              : 0
       total no. reference observed             : 0

Time elapsed: 0.00s


stats: Total number of observed variants   0
       Total number of unique variants     0

Time elapsed: 0.00s

Traceback (most recent call last):
  File "/opt/bcbio/anaconda/bin/bcbio_nextgen.py", line 221, in <module>
    install.upgrade_bcbio(kwargs["args"])
  File "/opt/bcbio/anaconda/lib/python3.6/site-packages/bcbio/install.py", line 106, in upgrade_bcbio
    upgrade_bcbio_data(args, REMOTES)
  File "/opt/bcbio/anaconda/lib/python3.6/site-packages/bcbio/install.py", line 348, in upgrade_bcbio_data
    args.cores, ["ggd", "s3", "raw"])
  File "/tmp/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/genomes.py", line 354, in install_data_local
    _prep_genomes(env, genomes, genome_indexes, ready_approaches, data_filedir)
  File "/tmp/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/genomes.py", line 480, in _prep_genomes
    retrieve_fn(env, manager, gid, idx)
  File "/tmp/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/genomes.py", line 850, in _install_with_ggd
    ggd.install_recipe(os.getcwd(), env.system_install, recipe_file, gid)
  File "/tmp/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/ggd.py", line 30, in install_recipe
    recipe["recipe"]["full"]["recipe_type"], system_install)
  File "/tmp/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/ggd.py", line 62, in _run_recipe
    subprocess.check_output(["bash", run_file])
  File "/opt/bcbio/anaconda/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/opt/bcbio/anaconda/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['bash', '/opt/bcbio/genomes/Hsapiens/hg19/txtmp/ggd-run.sh']' returned non-zero exit status 255.

```

**Comment by chapmanb** [comment]
Luca;
Apologies about the issue. gnomad 2.1 has been a difficult integration due to the large file sizes and challenges in filtering down the annotations. This is still a work in progress and @naumenko-sa has been working through potential solutions (https://github.com/chapmanb/cloudbiolinux/issues/298#issuecomment-489142860)

**Comment by lbeltrame** [comment]
Is there a way to bypass this just in case, what is the `datatarget` that pulls this in? Because even after removing `tmpbcbio-install` and removing `--datatarget cadd` it still tries to download the gnomad exomes (and fails, see the linked bug report).

**Comment by naumenko-sa** [comment]
Hi @lbeltrame !

Thanks for reporting the issue and sorry that you have trouble updating gnomad to 2.1
I was able to reproduce and pushed a fix - now it works on a small file for me.

You may replace the recipe in your tmpbcbio-install/cloudbiolinux/ggd-recipes/hg19
and rerun bcbio_nextgen.py upgrade .. command
The new recipes for exome and genome are here
https://github.com/naumenko-sa/cloudbiolinux/tree/master/ggd-recipes/hg19

Or update after PR gets merged:
https://github.com/chapmanb/cloudbiolinux/pull/300

Let me know if you see any other issues.

@chapmanb, 
gnomad2.1 genome processing is very slow. I'm generating grch37 for more than a week for now. 
I propose just to generate them one time for grch37, hg38, and hg19 (I'm doing that) 
and push somewhere in the cloud (cloudbiolinux bucket?)

Sergey

**Comment by chapmanb** [comment]
Sergey and Luca;
Thank you both for the reports and the helpful fixes. Hopefully the new updates get things running for Luca. For the longer term I'm agreed that the processing times here seem like a no-go and hosting these is probably the best option. The downside is always the hosting/download costs but I don't see a reasonable workaround so let's do it. Sergey, do you have prepared files on AWS/S3 or somewhere else? If you're having trouble getting it finished, we could also run this separately and get them uploaded, just trying to use your head start and knowledge of the problem. Thanks again.

**Comment by naumenko-sa** [comment]
Hi Brad!
I'm processing the files on a local cluster. Now with per chromosome processing in parallel, I hope to get all 6 finished in 2-3 days. Maybe we can ask the Ensembl team to host the resulting merged files? It would be beneficial for many of their users, not just for bcbio and it is not a huge addition to their ftp server. Do you know by any chance whom we may contact in Ensembl?  SN

**Comment by chapmanb** [comment]
Sergey;
That's a great idea. Hopefully Ensembl would have interest in hosting this. I don't have any specific contacts but I think posting on the development list (https://lists.ensembl.org/mailman/listinfo/dev_ensembl.org) would be a good place to start. Thanks again.

**Comment by naumenko-sa** [comment]
Hi Brad!

1. Could you please merge
https://github.com/chapmanb/cloudbiolinux/pull/302
I found another issue - the order of vcf files when merging.
2. concatenated exome files are ready for hg38, hg19, grch37. They are 3.6G each.
3. gnomad genome per chr downloading and processing is running
4. Linking a thread in Ensembl mail list
https://lists.ensembl.org/pipermail/dev_ensembl.org/2019-May/013346.html

SN

**Comment by chapmanb** [comment]
Sergey;
Thanks much for the fixes, being on top of these, and the messages to the Ensembl mailing list. Fingers crossed they'll be able to host and then we can transfer this to a download based install, with the recipes just kept for future updates (which will hopefully be infrequent).

**Comment by naumenko-sa** [comment]
Hi Brad and everyone who follows gnomad2.1 drama!

* Added correct merging for genome recipes and helper scripts to process and merge gnomad vcf locally
https://github.com/chapmanb/cloudbiolinux/pull/303. Please merge. 
* gnomad.vcf.gz for grch37 is ready. It is 55G (compared to 370G for original chr vcfs). It took 2 weeks to run the recipe with 1 thread. In parallel, with the helper scripts, it takes ~1day
* gnomad.vcf.gz for hg38 and hg19 are almost ready, hopefully, <1day left
* let me know if anybody has ftp server and wants to host these files, I'd be happy to upload.

SN

**Comment by naumenko-sa** [comment]
Hi Brad!

Unfortunately, Ensembl does not want to host these files.
https://lists.ensembl.org/pipermail/dev_ensembl.org/2019-May/013348.html

Do you think reaching gnomad team might help?

Or we might just roll back to gnomad2.0.1 - not ideal, but it is better to have working gnomad by default rather than not working gnomad. The advanced users may run processing scripts provided outside of cloudbiolinux to upgrade their gnomad to 2.1.

Sergey

**Comment by naumenko-sa** [comment]
Hi Brad!
Slim files for gnomad2.1 genome (55G) and exome (5G) for hg19/hg38/grch37 are ready for uploading.
The final portion of helper scripts:
https://github.com/chapmanb/cloudbiolinux/pull/305
S.

**Comment by chapmanb** [comment]
Sergey;
Thanks much for being on top of this and getting these all finished. I'm bummed about the Ensembl response and think your idea of asking the Gnomad team might work. Another idea is the genome in a bottle community (https://groups.google.com/forum/#!forum/giab-analysis-team). While this is less directly related there, it's a useful resource for variant interpretation and filtering, so might be of interest. Fingers crossed that one of these options works.

**Comment by naumenko-sa** [comment]
Wrote to Gnomad team:
https://gnomad.broadinstitute.org/contact

**Comment by naumenko-sa** [comment]
NCBI (GIAB) also cannot host the files. Can anybody kindly ping the Gnomad team, if you have contacts?
SN

**Comment by roryk** [comment]
Hi Sergey,

I tried tweeting at Dan, let's see if that works.

**Comment by ameynert** [comment]
Is it correct that release 1.1.5 will not properly process hg38 gnomad 2.1? I need to upgrade for other reasons, what's the easiest way to get just the updated ggd recipes without having to upgrade to development?

**Comment by naumenko-sa** [comment]
Hi @ameynert !

bcbio1.1.5 uses new recipes from cloudbiolinux with gnomad2.1. I think they are working, just the processing is very slow - it might take 1 week to process gnomad genomes.

You may use helper scripts and this method to produce the merged gnomad 2.1 file:
https://github.com/chapmanb/cloudbiolinux/blob/master/doc/hacking.md#testing-a-ggd-recipe
https://github.com/chapmanb/cloudbiolinux/tree/master/ggd-recipes/hg38

Also if you can host the resulting gnomad2.1. vcf (55G) for the community, let us know. The file is ready and I can upload it.

Sergey

**Comment by roryk** [comment]
Thanks-- do you know if we have a way to disable the update to 2.1? When I pull in a --data update, I get 2.1 and it is more than I bargained for.

**Comment by ameynert** [comment]
Thanks @naumenko-sa - my installer is on chr4 after <24 hours. I'll report back when it's finished or crashed. I will talk to our IT about the possibility of hosting the 55GB slim file.

**Comment by lbeltrame** [comment]
Related to @roryk 's comment, how can I tell bcbio NOT to download the gnomAD genome files? I already built the exomes and that would be enough. However, with every `--data` I get prompted to get the genomes.

Another fresh install I'm doing is not doing that (so far), so I assume there must be something somewhere.

**Comment by roryk** [comment]
@naumenko-sa told me how to do it, but now I have forgotten. Sergey, could you remind us?

**Comment by naumenko-sa** [comment]
Hi @lbeltrame  and @roryk!

You may try to start the upgrade, wait till tmpbcbio-install/cloudbiolinux/ggd-recipes folder is created,
then terminate the upgrade. Modify tmpbcbio-install/cloudbiolinux/ggd-recipes/hg19/gnomad.yaml (just comment downloading and processing, or put there an old recipe). Restart the upgrade. 

So it has been 2 months since we started trying to find a place to host for gnomad2.1. What should we do? Revert changes back to 2.0.1? @mbootwalla, any luck with hosting at your institution?

Sergey

**Comment by roryk** [comment]
Thanks, I think it makes sense to revert back while we figure out how to handle the 2.1 changes. @chapmanb what do you think?

**Comment by roryk** [comment]
https://raw.githubusercontent.com/chapmanb/cloudbiolinux/b6605021d7f9c9a36ae9c965819fe7ffa13c309d/ggd-recipes/hg38/gnomad_exome.yaml

and

https://raw.githubusercontent.com/chapmanb/cloudbiolinux/b6605021d7f9c9a36ae9c965819fe7ffa13c309d/ggd-recipes/hg38/gnomad.yaml

are the most recent 2.0.1 versions.

**Comment by chapmanb** [comment]
Luca -- if you want to skip gnomad installs/updates on an existing bcbio installation you can remove it from the datatarget options in `bcbio/config/install-params.yaml`, and then it will no longer try to grab it.

Sergey and Rory -- thanks for all the patience on this. My only other suggestion besides reverting is that we could host on S3 if we're cool paying the download costs. That might be a conversation to have since I know you spent a lot of time getting it prepared and it's a shame to have to go back.

**Comment by lbeltrame** [comment]
It doesn't seem to work. I commented it out and it's still trying to download the hg38 exomes (which I already have in place).

**Comment by naumenko-sa** [comment]
@lbeltrame . Could you please try to set `gnomad_exome,2.1` in your `bcbio/genomes/Hsapiens/hg19/versions.csv`. If you have 2.05 there, then bcbio is trying to install the new recipe which is 2.1

@chapmanb, @roryk. Gnomad now provides merged vcf files for all chromosomes.
https://storage.googleapis.com/gnomad-public/release/2.1.1/vcf/genomes/gnomad.genomes.r2.1.1.sites.vcf.bgz
WGS = 460G, EXOMES = 59G.

It solves the hosting problem, and now I can simplify the installation script. Maybe, it is not that bad? I tried a couple of connections now, including O2, it promised downloading WGS in 2-7h.

SN

**Comment by mbootwalla** [comment]
Hi Sergey and bcbio team,

I am working with my supervisor on pushing this through. I will get back to
you guys in a couple of days and let you know where this stands. I
apologize for this taking so long.

Thanks,

Moiz

**Comment by roryk** [comment]
Hi everyone,

Great-- if we grab https://storage.googleapis.com/gnomad-public/release/2.1.1/vcf/exomes/gnomad.exomes.r2.1.1.sites.vcf.bgz for the exomes and index it, is that enough for it to work as the gnomad_exome.vcf.gz file, or do we also need to do the fields_to_keep formatting and what not?

**Comment by naumenko-sa** [comment]
Hi Rory!

Sorry for the delay - could you please review and merge 3 recipes for gnomad exome?
https://github.com/chapmanb/cloudbiolinux/pull/322

SN

**Comment by roryk** [comment]
Thanks Sergey, that worked great!
